apps/emqx_bridge_datalayers/test/emqx_bridge_datalayers_arrow_SUITE.erl

```diff
@@ -0,0 +1,461 @@
+            {ok, Result};
+        {Ref, Result} ->
+            {ok, Result}
+    after Timeout ->
+        timeout
+    end.
+
+query_by_clientid(ClientId, Config) ->
+    SQL = io_lib:format("SELECT * FROM mqtt_arrow.common_test WHERE clientid = '~s'", [ClientId]),
+    case exec_sql(bin(SQL), Config) of
+        {ok, Rows} ->
+            Rows;
+        {error, Reason} ->
+            error({bad_response, Reason})
+    end.
+
+assert_persisted_data(Expected, [PersistedData | _]) ->
+    ?assertEqual(Expected, PersistedData).
+
+connector_id(Config) ->
+    Name = ?config(connector_name, Config),
+    emqx_connector_resource:resource_id(<<"datalayers">>, Name).
+
+ensure_database(Config) ->
+    exec_sql(?create_database, Config).
+
+ensure_table(Config) ->
+    exec_sql(?create_table, Config).
+
+id(Type, Name) ->
+    emqx_bridge_v2_testlib:lookup_chan_id_in_conf(#{
+        kind => action,
+        type => Type,
+        name => Name
+    }).
+
+%%------------------------------------------------------------------------------
+%% Testcases
+%%------------------------------------------------------------------------------
+
+t_start_ok(Config) ->
+    QueryMode = ?config(query_mode, Config),
+    ?assertMatch(
+        {ok, _},
+        create_bridge(Config)
+    ),
+    ClientId = random_clientid_(),
+    Topic = <<"t/", (atom_to_binary(?FUNCTION_NAME))/binary>>,
+    Payload = #{msg => <<"Hello, Arrow Flight SQL!">>},
+    #{id := Id, timestamp := TimeStamp} = Msg = make_message(ClientId, Topic, Payload),
+
+    Request = {emqx_bridge_v2_testlib:bridge_id(Config), Msg},
+    case QueryMode of
+        sync ->
+            query_resource(Config, Request);
+        async ->
+            {_, Ref} = query_resource_async(Config, Request),
+            {ok, Res} = receive_result(Ref, 2_000),
+            Res
+    end,
+    ct:sleep(1500),
+    PersistedData = query_by_clientid(ClientId, Config),
+
+    Expected = [
+        _TsStr = iolist_to_binary(
+            calendar:system_time_to_rfc3339(TimeStamp, [{unit, millisecond}, {offset, "Z"}])
+        ),
+        Id,
+        ClientId,
+        Topic,
+        _Qos = <<"0">>,
+        emqx_utils_json:encode(Payload)
+    ],
+    assert_persisted_data(Expected, PersistedData),
+    ok.
+
+t_start_stop(Config) ->
+    ok = emqx_bridge_v2_testlib:t_start_stop(Config, "arrow_flight_sql_client_stopped").
+
+t_get_status(Config) ->
+    ProxyPort = ?config(proxy_port, Config),
+    ProxyHost = ?config(proxy_host, Config),
+    ProxyName = ?config(proxy_name, Config),
+    {ok, _} = create_bridge(Config),
+    ConnectorId = emqx_connector_resource:resource_id(datalayers, ?config(connector_name, Config)),
+    ?assertEqual({ok, connected}, emqx_resource_manager:health_check(ConnectorId)),
+    emqx_common_test_helpers:with_failure(down, ProxyName, ProxyHost, ProxyPort, fun() ->
+        ?assertEqual({ok, disconnected}, emqx_resource_manager:health_check(ConnectorId))
+    end),
+    ok.
```

Comment:
See `emqx_bridge_v2_testlib:t_on_get_status`, which does this in a more standardized way.